src/System.Text.JsonLab.Dynamic/System/Text/Json/Deserializer.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace System.Text.JsonLab
+{
+    public partial class JsonDynamicObject : DynamicObject, IBufferFormattable
+    {
+        private static class Deserializer<T>
+        {
+            private static readonly Func<T> _createInstance = CreateDelegate();
+
+            private static Func<T> CreateDelegate()
+            {
+                DynamicMethod method = new DynamicMethod("CreateInstanceDynamicMethod", typeof(T), null, restrictedSkipVisibility: true);
+
+                // GetConstructors() is only available on netstandard 2.0
+                IEnumerable<ConstructorInfo> ctors = typeof(T).GetTypeInfo().DeclaredConstructors;
+                ConstructorInfo constructor = default;
+                foreach (ConstructorInfo ci in ctors)
+                {
+                    constructor = ci;
+                    break;
```

Comment:
So you just pick the first ctor regardless of parameters?

Reply:
I will add a comment, TODO.

I understand it doesn't cover most scenarios (for instance if the type to deserialize has custom, non-default constructors). Wanted to start with supporting deserializing of simple types first.